### ISSUE TITLE
errors

### ISSUE THREAD

**Issue by swvalenti194** [jishi/node-sonos-http-api#661]
I installed however you can see all the errors....not sure what I did or am doing wrong?  I can get the sonos module to show up on magic mirror but just says loading music....any help please?

> sonos-http-api@1.6.5 start /home/pi/node-sonos-http-api
> node server.js

/home/pi/node-sonos-http-api/node_modules/music-metadata/lib/index.js:15
function parseStream(stream, mimeType, options = {}) {
                                               ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:414:25)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/home/pi/node-sonos-http-api/lib/helpers/file-duration.js:1:81)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)

npm ERR! Linux 4.14.71-v7+
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v4.2.1
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE
npm ERR! sonos-http-api@1.6.5 start: `node server.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sonos-http-api@1.6.5 start script 'node server.js'.
npm ERR! This is most likely a problem with the sonos-http-api package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls sonos-http-api
npm ERR! There is likely additional logging output above.

**Comment by jishi** [comment]
Seems like some dependency doesn't work in node 4. I have to figure out
which one it is and pin it to an older version perhaps. Easiest remedy for
you would be to update node.js to 6 or higher, if possible.

On Nov 10, 2018 20:31, "swvalenti194" <notifications@github.com> wrote:

I installed however you can see all the errors....not sure what I did or am
doing wrong? I can get the sonos module to show up on magic mirror but just
says loading music....any help please?

sonos-http-api@1.6.5 start /home/pi/node-sonos-http-api
node server.js

/home/pi/node-sonos-http-api/node_modules/music-metadata/lib/index.js:15
function parseStream(stream, mimeType, options = {}) {
^

SyntaxError: Unexpected token =
at exports.runInThisContext (vm.js:53:16)
at Module._compile (module.js:414:25)
at Object.Module._extensions..js (module.js:442:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:311:12)
at Module.require (module.js:366:17)
at require (module.js:385:17)
at Object. (/home/pi/node-sonos-http-api/lib/helpers/file-duration.js:1:81)
at Module._compile (module.js:435:26)
at Object.Module._extensions..js (module.js:442:10)

npm ERR! Linux 4.14.71-v7+
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v4.2.1
npm ERR! npm v2.14.7
npm ERR! code ELIFECYCLE
npm ERR! sonos-http-api@1.6.5 start: node server.js
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sonos-http-api@1.6.5 start script 'node server.js'.
npm ERR! This is most likely a problem with the sonos-http-api package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR! node server.js
npm ERR! You can get their info via:
npm ERR! npm owner ls sonos-http-api
npm ERR! There is likely additional logging output above.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/jishi/node-sonos-http-api/issues/661>, or mute the
thread

**Comment by swvalenti194** [comment]
I updated to the latest node version 11 and this is what I get...just hangs on listening...


pi@raspberrypi:~/node-sonos-http-api $ npm start

> sonos-http-api@1.6.5 start /home/pi/node-sonos-http-api
> node server.js

2018-11-11T14:10:58.109Z INFO Presets loaded: { example:
   { players:
      [ { roomName: 'Bathroom', volume: 10 },
        { roomName: 'Kitchen', volume: 10 },
        { roomName: 'Office', volume: 10 },
        { roomName: 'Bedroom', volume: 10 },
        { roomName: 'TV Room', volume: 15 } ],
     playMode: { shuffle: true, repeat: 'all', crossfade: false },
     pauseOthers: false } }
2018-11-11T14:11:00.341Z INFO http server listening on 0.0.0.0 port 5005

**Comment by jishi** [comment]
I use pm2 which is fairly easy to setup. See https://github.com/jishi/node-sonos-http-api/issues/20

What do you mean by hiding a zone? If it's a magic mirror thing, you'd have to check with the person who wrote that module. There is no concept of hiding zones in this API.

**Comment by swvalenti194** [comment]
Thanks, and I mean on the MM Sonos shows the 3 zones that I have (portable, Scott's, Sarah's)  I'd like to hide portable from showing music if possible?

**Comment by swvalenti194** [comment]
Im trying to follow #20 but not working.  I have pm2 installed already as I use it for another service.  

pi@raspberrypi:~ $ pm2 start npm -- start
[PM2] Starting /usr/local/bin/npm in fork_mode (1 instance)
[PM2] Done.
ââââââââ¬âââââ¬âââââââ¬âââââââââ¬ââââââââââ¬ââââââ¬âââââââââ¬ââââââââââââ
â Name â id â mode â status â âº       â cpu â memory â
ââââââââ¼âââââ¼âââââââ¼âââââââââ¼ââââââââââ¼ââââââ¼âââââââââ¼ââââââââââââ¤
â mm   â 0  â N/A  â fork   â stopped â 6   â 0%     â 0 B       â
â npm  â 1  â N/A  â fork   â online  â 0   â 0%     â 15.9 MB   â
ââââââââ´âââââ´âââââââ´âââââââââ´ââââââââââ´ââââââ´âââââââââ´ââââââââââââ
 Use `pm2 show <id|name>` to get more details about an app
pi@raspberrypi:~ $ pm2 save
[PM2] Saving current process list...
[PM2] Successfully saved in /home/pi/.pm2/dump.pm2
pi@raspberrypi:~ $ cd /home/pi/node-sonos-http-api
pi@raspberrypi:~/node-sonos-http-api $ cd
pi@raspberrypi:~ $ pm2 startup raspberry
[PM2] Init System found: systemd
-----------------------------------------------------------
 PM2 detected systemd but you precised raspberry
 Please verify that your choice is indeed your init system
 If you arent sure, just run : pm2 startup
-----------------------------------------------------------
[PM2] To setup the Startup Script, copy/paste the following command:
sudo env PATH=$PATH:/usr/local/bin /usr/lib/node_modules/pm2/bin/pm2 startup raspberry -u pi --hp /home/pi
pi@raspberrypi:~ $ sudo env PATH=$PATH:/usr/local/bin pm2 startup raspberry -u pi
[PM2] Init System found: systemd
-----------------------------------------------------------
 PM2 detected systemd but you precised raspberry
 Please verify that your choice is indeed your init system
 If you arent sure, just run : pm2 startup
-----------------------------------------------------------
/usr/lib/node_modules/pm2/lib/API/Startup.js:304
      throw new Error('Unknown platform / init system name');
      ^

Error: Unknown platform / init system name
    at API.CLI.startup (/usr/lib/node_modules/pm2/lib/API/Startup.js:304:13)
    at Command.<anonymous> (/usr/lib/node_modules/pm2/bin/pm2:752:9)
    at Command.listener (/usr/lib/node_modules/pm2/node_modules/commander/index.js:315:8)
    at Command.emit (events.js:182:13)
    at Command.parseArgs (/usr/lib/node_modules/pm2/node_modules/commander/index.js:651:12)
    at Command.parse (/usr/lib/node_modules/pm2/node_modules/commander/index.js:474:21)
    at Timeout._onTimeout (/usr/lib/node_modules/pm2/bin/pm2:220:15)
    at listOnTimeout (timers.js:324:15)
    at processTimers (timers.js:268:5)

**Comment by swvalenti194** [comment]
Everything works just fine but I can't for the life of me figure out how start the api on boot....please help...even though I referenced the others

**Comment by swvalenti194** [comment]
I have to keep going to cd node-sonos-http-api and then npm start which is less than annoying

**Comment by jishi** [comment]
Yoy shouldn't use npm start script if you run it with pm2. Just reference
the starting script

pm2 server.js

**Comment by swvalenti194** [comment]
can you dumb it down what you said as i am a newb

**Comment by jishi** [comment]
And if you already run pm2 there is no need to run the startup
initialization. That is a one time thing for the pm2 manager. Any processes
you start with pm2 will be controlled by the same i it startup.

**Comment by swvalenti194** [comment]
ok then why isnt it starting upon reboot?  sorry for the dumbness

**Comment by jishi** [comment]
cd into the sonos-http-api folder.

Run: pm2 start server.js

Run: pm2 save

That should be it, if you already had pm2 installed and working at boot.

**Comment by swvalenti194** [comment]
you are the man that worked thank you kindly sir
